api/prometheus/v1/api.go

```diff
@@ -670,11 +680,20 @@ type apiClient struct {
 	api.Client
 }
 
+// TODO: create a wrapper for multiple errors, this way we can type switch on these
+// and do what we need to
```

Comment:
why not do it as part of this PR?

also the comment formating should be something like:
```
// Warnings is returned when .......
// TODO: create a wrapper for multiple errors, this way we can type switch on these
// and do what we need to.
```

Reply:
WIP; so far I've been given some contradicting or unclear direction. This will get cleaned up before we merge.

Reply:
This is an interface question. If we want to always return `returnValue, warnings, error` there is no need for this. If we want to instead keep the `returnValue, error` then we need to make that error switch-able.

So, do we want to *add* warnings to the return signature? Or try to type-switch it out?